.github/workflows/stale-repos.yml

```diff
@@ -1,19 +1,35 @@
 on:
   schedule:
-  - cron: "0 */6 * * *" # every 6 hours
+    - cron: "0 */6 * * *" # every 6 hours
+  pull_request:
+    paths:
+      - 'configs/*.json'
+
+name: Check Stale Repos
 
 jobs:
   matrix:
     name: Find stale repositories
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - id: repos
+        name: find repositories to check
+        run: |
+          repos="$(jq '{"\(input_filename)":(.repositories | map(.target))}' configs/*.json | jq -sc 'add')"
+          if [[ '${{ github.event_name }}' == 'pull_request' ]]; then
+            git fetch origin ${{ github.event.pull_request.base.sha }}
+            git reset --hard ${{ github.event.pull_request.base.sha }}
+            base="$(jq '{"\(input_filename)":(.repositories | map(.target))}' configs/*.json | jq -sc 'add')"
```

Comment:
So much jq magic! ;)
A comment explaining what this does would be nice. Same for the next line.

Reply:
Also, nit: you can do this in one command `jq -nc '[ inputs | ... ] | add`

Reply:
Good points - done & done 👍 🙇